Only in PHPToolkit_2014_1: HvTweaks.patch
diff -ur PHPToolkit_2014_1.orig/PHPToolkit/NetSuiteService.php PHPToolkit_2014_1/PHPToolkit/NetSuiteService.php
--- PHPToolkit_2014_1.orig/PHPToolkit/NetSuiteService.php	2014-05-06 13:01:50.000000000 -0600
+++ PHPToolkit_2014_1/PHPToolkit/NetSuiteService.php	2014-05-29 17:59:49.135186745 -0600
@@ -1,5 +1,5 @@
 <?php
-
+namespace NetSuite\WebServices;
 require_once "NSPHPClient.php";
 
 if (!class_exists("RecordType")) {
@@ -127674,8 +127674,8 @@
 	 * @param string $wsdl WSDL location for this service
 	 * @param array $options Options for the SoapClient
 	 */
-	public function __construct($wsdl=null, $options=array()) {
-		parent::__construct($wsdl, $options);
+	public function __construct(NSconfig $nsconfig, $wsdl=null, $options=array()) {
+		parent::__construct($nsconfig, $wsdl, $options);
 	}
 
 	/**
@@ -128244,4 +128244,4 @@
 
 }}
 
-?>
\ No newline at end of file
+?>
diff -ur PHPToolkit_2014_1.orig/PHPToolkit/NSconfig.php PHPToolkit_2014_1/PHPToolkit/NSconfig.php
--- PHPToolkit_2014_1.orig/PHPToolkit/NSconfig.php	2014-05-06 14:58:56.000000000 -0600
+++ PHPToolkit_2014_1/PHPToolkit/NSconfig.php	2014-05-29 16:59:16.941720720 -0600
@@ -1,9 +1,15 @@
 <?php
-$nsendpoint = "2014_1";
-$nshost = "https://webservices.netsuite.com";
+namespace NetSuite\WebServices;
+
+class  NSconfig {
+    public $endpoint = "2014_1";
+    public $host = "https://webservices.netsuite.com";
+    public $email = "jDoe@netsuite.com";
+    public $password = "mySecretPwd";
+    public $role = "3";
+    public $account = "MYACCT1";
+    public $debuginfo = null;
+    public $logging = false;
+}
 
-$nsemail = "jDoe@netsuite.com";
-$nspassword = "mySecretPwd";
-$nsrole = "3";
-$nsaccount = "MYACCT1";
 ?>
Only in PHPToolkit_2014_1/PHPToolkit: nslog
diff -ur PHPToolkit_2014_1.orig/PHPToolkit/NSPHPClient.php PHPToolkit_2014_1/PHPToolkit/NSPHPClient.php
--- PHPToolkit_2014_1.orig/PHPToolkit/NSPHPClient.php	2014-05-06 15:17:26.000000000 -0600
+++ PHPToolkit_2014_1/PHPToolkit/NSPHPClient.php	2014-05-29 18:09:59.951875726 -0600
@@ -1,6 +1,14 @@
 <?php
+namespace NetSuite\WebServices;
 
-require "NSconfig.php";
+function class_exists($class_name, $autoload = true)
+{
+    if(strstr(__NAMESPACE__, 0, strlen(__NAMESPACE__)) === FALSE)
+    {
+        $class_name = __NAMESPACE__ . '\\' . $class_name;
+    }
+    return \class_exists($class_name, $autoload);
+}
 
 function arrayValuesAreEmpty ($array)
 {
@@ -138,7 +146,7 @@
 function cleanUpNamespaces($xml_root)
 {
     $xml_root = str_replace('xsi:type', 'xsitype', $xml_root);
-    $record_element = new SimpleXMLElement($xml_root);
+    $record_element = new \SimpleXMLElement($xml_root);
 
     foreach ($record_element->getDocNamespaces() as $name => $ns)
     {
@@ -148,11 +156,11 @@
         }
     }
 
-    $record_element = new SimpleXMLElement($xml_root);
+    $record_element = new \SimpleXMLElement($xml_root);
 
     foreach($record_element->children() as $field)
     {
-        $field_element = new SimpleXMLElement($field->asXML());
+        $field_element = new \SimpleXMLElement($field->asXML());
 
         foreach ($field_element->getDocNamespaces() as $name2 => $ns2)
         {
@@ -175,19 +183,26 @@
     private $userequest = true;
     protected $classmap = null;
     public $generated_from_endpoint = "";
+    protected $config = null;
+    public $logging = false;
 
 
-    protected function __construct($wsdl=null, $options=array()) {
-        global $nshost, $nsendpoint;
-        global $nsaccount, $nsemail, $nsrole, $nspassword;
-        global $debuginfo;
-
+    protected function __construct(NSconfig $nsconfig, $wsdl=null, $options=array()) {
+        $nshost = $nsconfig->host;
+        $nsendpoint = $nsconfig->endpoint;
+        $nsaccount = $nsconfig->account;
+        $nsemail = $nsconfig->email;
+        $nsrole = $nsconfig->role;
+        $nspassword = $nsconfig->password;
+        $debuginfo = $nsconfig->debuginfo;
+        $this->logging = $nsconfig->logging;
+        
         if (!isset($wsdl)) {
              if (!isset($nshost)) {
-                throw new Exception('Webservice host must be specified');
+                throw new \Exception('Webservice host must be specified');
              }
              if (!isset($nsendpoint)) {
-                throw new Exception('Webservice endpoint must be specified');
+                throw new \Exception('Webservice endpoint must be specified');
              }
              $wsdl = $nshost . "/wsdl/v" . $nsendpoint . "_0/netsuite.wsdl";
         }
@@ -210,6 +225,10 @@
             trigger_error($endpoint_warning, E_USER_WARNING);
         }
 
+        foreach($this->classmap as $key => $value){
+            $this->classmap[$key] = __NAMESPACE__ . "\\" . $value;
+        }
+
         $options['classmap'] = $this->classmap;
         $options['trace'] = 1;
         $options['connection_timeout'] = 5;
@@ -237,9 +256,13 @@
         $options['user_agent'] =  $httpheaders;
         $this->setPassport($nsaccount, $nsemail, $nsrole, $nspassword);
 
-        $this->client = new SoapClient($wsdl, $options);
+        $this->client = new \SoapClient($wsdl, $options);
     }
 
+    public function setLocation($location) {
+        $this->client->__setLocation($location);
+    }
+    
     public function setPassport($nsaccount, $nsemail, $nsrole, $nspassword) {
         $this->passport = new Passport();
         $this->passport->account = $nsaccount;
@@ -283,7 +306,7 @@
     }
 
     public function addHeader($header_name, $header) {
-        $this->soapHeaders[$header_name] = new SoapHeader("ns", $header_name, $header);
+        $this->soapHeaders[$header_name] = new \SoapHeader("ns", $header_name, $header);
     }
     public function clearHeader($header_name) {
         unset($this->soapHeaders[$header_name]);
@@ -301,7 +324,7 @@
 
         $response = $this->client->__soapCall($operation, array($parameter), NULL, $this->soapHeaders);
 
-        if ( file_exists(dirname(__FILE__) . '/nslog') ) {
+        if ( $this->logging && file_exists(dirname(__FILE__) . '/nslog') && is_writable(dirname(__FILE__) . '/nslog') ) {
             // log the request and response into the nslog directory. Code taken from PHP toolkit
             // REQUEST
             $req = dirname(__FILE__) . '/nslog' . "/" . date("Ymd.His") . "." . milliseconds() . "-" . $operation . "-request.xml";
Only in PHPToolkit_2014_1.orig/samples: ass_invoice.php
